### PR TITLE
Drops action_filter support

### PR DESCRIPTION
**Pull request by bgreg**
This small change fixes the rails 5.1.2 error, and hard codes the `after_action` callback.

### CODE REVIEW

**Comment by tute** [comment]
Yep! Shall we update the Travis CI matrix to disallow unsupported Rails' versions, and do a major release? Thanks!

**Comment by tute** [comment]
Related with: https://github.com/merit-gem/merit/issues/274. We'll also want to update the gemspec.

**Comment by bgreg** [comment]
I'm looking at that travis build now.

**Comment by tute** [comment]
I'm fine ignoring most Hound comments in this PR. Thanks for your work!

**Comment by bgreg** [comment]
@tute What do you think about this case statement. Looks like it will be dead code too:

**Comment by tute** [comment]
> @tute What do you think about this case statement from the Gemfile. Looks like it will be dead code too:

Yep! We can delete the `rails` local variable, and use `version` instead. Love this.

**Comment by bgreg** [comment]
Cool, will do.  Also, there is something I don't understand about travis. Why is it running versions of rails other than what I have specified on this branch?  Is it _also_ using the master version of the `.travis.yml` file?

**Comment by tute** [comment]
> Cool, will do. Also, there is something I don't understand about travis. Why is it running versions of rails other than what I have specified on this branch? Is it also using the master version of the .travis.yml file?
> View changes

It looks like! Shouldn't be running 4.2. Let's ignore those for now.

**Comment by tute** [comment]
Don't forget it seems we can clean up some code in `test/dummy/config/environments/test.rb`.

**Comment by tute** [comment]
> I have no problem with this, given people on 5 can use merit ~2. But curious, what bugs were we hitting?

I'd rather keep 5 if we can. 5.0 and 5.1 are similar enough for us.

**Comment by bgreg** [comment]
I was getting 'ArgumentError: key must be 32 byte' when running against 5.0.0.0. Oddly enough I can't reproduce that again. I'll give travis another chance at ~> 5.0, and fix that test.rb file.

**Comment by bgreg** [comment]
Dropping that from the dummy env files causes:

```
config.eager_load is set to nil. Please update your config/environments/*.rb files accordingly:

  * development - set it to false
  * test - set it to false (unless you use a tool that preloads your test environment)
  * production - set it to true
```

But turning off warnings seems nice.  Sending another build

**Comment by bgreg** [comment]
I'm stuck on this failure right now: https://travis-ci.org/merit-gem/merit/jobs/254626719

**Comment by bgreg** [comment]
@tute  could use some input, running out of easy fix ideas for this:

```.............................................
Finished in 1.759557s, 25.5746 runs/s, 74.4506 assertions/s.
/home/travis/build/merit-gem/merit/vendor/bundle/ruby/2.4.0/gems/railties-5.1.0/lib/rails/test_unit/minitest_plugin.rb:9:in `aggregated_results': wrong number of arguments (given 1, expected 0) (ArgumentError)
	from /home/travis/build/merit-gem/merit/vendor/bundle/ruby/2.4.0/gems/minitest-5.10.2/lib/minitest.rb:597:in `report'
	from /home/travis/build/merit-gem/merit/vendor/bundle/ruby/2.4.0/gems/minitest-5.10.2/lib/minitest.rb:687:in `each'
	from /home/travis/build/merit-gem/merit/vendor/bundle/ruby/2.4.0/gems/minitest-5.10.2/lib/minitest.rb:687:in `report'
	from /home/travis/build/merit-gem/merit/vendor/bundle/ruby/2.4.0/gems/minitest-5.10.2/lib/minitest.rb:141:in `run'
	from /home/travis/build/merit-gem/merit/vendor/bundle/ruby/2.4.0/gems/railties-5.1.0/lib/rails/test_unit/minitest_plugin.rb:77:in `run'
	from /home/travis/build/merit-gem/merit/vendor/bundle/ruby/2.4.0/gems/minitest-5.10.2/lib/minitest.rb:63:in `block in autorun'
rake aborted!

```

What do you think?

**Comment by tute** [comment]
Apparently minitest 5.10.2 has an issue. We can specify versions minitest '~> 5.10', '!= 5.10.2', or later.

**Comment by bgreg** [comment]
Making progress. Just some trial and error finding the lowest rails 5 version to support.

**Comment by bgreg** [comment]
@tute if you agree, can you click resolve on that travis file?

**Comment by tute** [comment]
aha, that means your branch was born before latest master.

Can you please squash your commits together, and rebase them on top of latest master? I wrote instructions for this in https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history. That'd resolve the conflicts in the process. Let me know if I can help you do this.

Thanks for all the work! :)

File: Rakefile [hunk 1]
```diff
@@ -13,8 +13,8 @@ task default: :test
 Rake::TestTask.new(:test) do |t|
   t.libs << 'lib'
   t.libs << 'test'
-  t.pattern = 'test/**/*_test.rb'
-  t.verbose = true
+  t.pattern  = 'test/**/*_test.rb'
+  t.verbose  = true
```
Comment (tute) [inline Rakefile]: No need of the extra white space here

File: lib/merit/controller_extensions.rb [hunk 2]
```diff
@@ -4,11 +4,7 @@ module Merit
   # 'controller_path#action_name'
   module ControllerExtensions
     def self.included(base)
-      if base.respond_to? :after_action
-        base.after_action :log_and_process
-      else
-        base.after_filter :log_and_process
-      end
+      base.after_action :log_and_process
```
Comment (tute) [inline lib/merit/controller_extensions.rb]: Yes!

File: .travis.yml [hunk 3]
```diff
@@ -3,7 +3,9 @@ sudo: false
...
 
 rvm:
-  - 2.3.0
+  - 2.3.3
+  - 2.4.1
 
 env:
-  - RAILS_VERSION=5.0
+  - RAILS_VERSION=5.0.3
+  - RAILS_VERSION=5.1
```
Comment (tute) [inline .travis.yml]: There will be changes to merge from latest master, when you rebase you'll see them.

**Comment by bgreg** [comment]
@tute ok, git surgery done. I think this might be ready for a final review. Thanks for being responsive ! It's a lot more fun when someone is working with me.

**Comment by tute** [comment]
Thanks a lot! :)

**Comment by tute** [comment]
Thanks a lot! :)

**Pull request merged**